python/feldera/rest/client.py

```diff
@@ -0,0 +1,455 @@
+                program=program.get("code"),
+                description=program.get("description"),
+            ) for program in resp
+        ]
+
+    def get_program(self, name: str, with_code: bool = False) -> Program:
+        """
+        Get a program by name
+        :param name: The name of the program
+        :param with_code: If True, the program code will be included in the response
+        """
+        resp = self.http.get(
+            path=f"/programs/{name}",
+            params={
+                "with_code": with_code,
+            }
+        )
+
+        return Program(
+            name=resp.get("name"),
+            program=resp.get("code"),
+            description=resp.get("description"),
+            status=resp.get("status"),
+            version=resp.get("version"),
+        )
+
+    def compile_program(self, program: Program):
+        """
+        Compiles a SQL program
+        :param program: The program to compile
+        """
+        body = {
+            "code": program.program,
+            "description": program.description or "",
+        }
+
+        resp = self.http.put(
+            path=f"/programs/{program.name}",
+            body=body,
+        )
+        program.id = resp.get("program_id")
+        program.version = resp.get("version")
+
+        while True:
+            status = self.get_program(program.name).status
+
+            if status == "Success":
+                break
+            elif status != "Pending" and status != "CompilingRust" and status != "CompilingSql":
+                raise RuntimeError(f"Failed program compilation with status {status}")
+
+            logging.debug("still compiling %s, waiting for 5 more seconds", program.name)
+            time.sleep(5)
+
+    def delete_program(self, name: str):
+        """
+        Deletes a program by name
+        :param name: The name of the program
+        """
+
+        self.http.delete(
+            path=f"/programs/{name}",
+        )
+
+    def connectors(self) -> list[Connector]:
+        """
+        Get all connectors
+        """
+
+        resp = self.http.get(
+            path="/connectors",
+        )
+
+        return [
+            Connector(
+                name=connector.get("name"),
+                description=connector.get("description"),
+                config=connector.get("config"),
+                id=connector.get("connector_id"),
+            ) for connector in resp
+        ]
+
+    def get_connector(self, name: str) -> Connector:
+        """
+        Get a connector by name
+        :param name: The name of the connector
+        """
+
+        resp = self.http.get(
+            path=f"/connectors/{name}",
+        )
+
+        return Connector(
+            name=resp.get("name"),
+            description=resp.get("description"),
+            config=resp.get("config"),
+            id=resp.get("connector_id"),
+        )
+
+    def create_connector(self, connector: Connector):

```

Comment:
Possibly boolean parameter `overwrite` to differentiate between post and put?

Reply:
Also applies to the other create functions

Reply:
I thought about this, but felt like rarely would there be cases where post is desirable. I might be wrong.